chef/cookbooks/barclamp/libraries/barclamp_library.rb

```diff
@@ -264,6 +264,15 @@ def self.unclaimed(node, include_mounted=false)
           end
         end
 
+        # return the disks that are unclaimed and prepared for given role (or with no role)
+        def self.prepared_for(node,role)
+          disk_roles = node["crowbar_wall"].disk_roles || {}
```

Comment:
Really doing such a mixed way of accessing attributes? I would prefer something unique like ```node["crowbar_wall"]["disk_roles"]```.